src/main/java/site/coach_coach/coach_coach_server/userrecord/service/UserRecordService.java

```diff
@@ -76,4 +76,17 @@ private LocalDate validateAndConvertToLocalDate(String date) {
 			throw new InvalidInputException(ErrorMessage.INVALID_VALUE);
 		}
 	}
+
+	public UserRecord getUserRecordForCompleteCategory(Long userId, LocalDate recordDate) {
+		if (userRecordRepository.existsByRecordDateAndUser_UserId(recordDate, userId)) {
+			return userRecordRepository.findByRecordDateAndUser_UserId(recordDate, userId);
+		} else {
+			User user = userRepository.findById(userId).orElseThrow(InvalidUserException::new);
+			UserRecord userRecord = UserRecord.builder()
+				.user(user)
+				.recordDate(recordDate)
+				.build();
+			return userRecordRepository.save(userRecord);
+		}
```

Comment:
`exists`와 `find`로 존재 여부를 두번 확인하는 것이 불필요한 중복 코드라는 생각이 들어서요... 
```suggestion
		return userRecordRepository.findByRecordDateAndUser_UserId(recordDate, userId)
        .orElseGet(() -> {
            User user = userRepository.findById(userId).orElseThrow(InvalidUserException::new);
            UserRecord userRecord = UserRecord.builder()
                .user(user)
                .recordDate(recordDate)
                .build();
            return userRecordRepository.save(userRecord);
        });
```
요롷게 바꿔볼 수도 있을 것 같아요!

Reply:
orElseGet 기능이 딱 필요했었는데 바로 코드가 간단해져서 좋네요!